applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp

```diff
@@ -0,0 +1,228 @@
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Mohammad R. Hashemi
+//
+
+// Project includes
+#include "utilities/variable_utils.h"
+#include "includes/global_pointer_variables.h"
+#include "utilities/parallel_utilities.h"
+#include "factories/linear_solver_factory.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+#include "distance_smoothing_process.h"
+
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+template< unsigned int TDim, class TSparseSpace, class TDenseSpace, class TLinearSolver>
+DistanceSmoothingProcess<TDim, TSparseSpace, TDenseSpace, TLinearSolver>::DistanceSmoothingProcess(
+    ModelPart& rModelPart,
+    typename TLinearSolver::Pointer plinear_solver)
+    : Process(),
+      mrModelPart(rModelPart)
+{
+    // Generate an auxilary model part and populate it by elements of type DistanceSmoothingElement
+    CreateAuxModelPart();
+
+    auto p_builder_solver = Kratos::make_shared<ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver> >(plinear_solver);
+
+    CreateSolutionStrategy(plinear_solver, p_builder_solver);
+}
+
+template< unsigned int TDim, class TSparseSpace, class TDenseSpace, class TLinearSolver>
+DistanceSmoothingProcess<TDim, TSparseSpace, TDenseSpace, TLinearSolver>::DistanceSmoothingProcess(
+    ModelPart& rModelPart,
+    Parameters& rParameters)
+    : Process(),
+      mrModelPart(rModelPart)
+{
+    // Generate an auxilary model part and populate it by elements of type DistanceSmoothingElement
+    CreateAuxModelPart();
+
+    typename TLinearSolver::Pointer plinear_solver = LinearSolverFactory<TSparseSpace, TDenseSpace>().Create(
+        rParameters["linear_solver_settings"]);
+
+    auto p_builder_solver = Kratos::make_shared<ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver> >(plinear_solver);
+
+    CreateSolutionStrategy(plinear_solver, p_builder_solver);
+}
+
+template< unsigned int TDim, class TSparseSpace, class TDenseSpace, class TLinearSolver>
+DistanceSmoothingProcess<TDim, TSparseSpace, TDenseSpace, TLinearSolver>::DistanceSmoothingProcess(
+    Model &rModel,
+    Parameters &rParameters)
+    : Process(),
+      mrModelPart(rModel.GetModelPart(rParameters["model_part_name"].GetString()))
+{
+    // Generate an auxilary model part and populate it by elements of type DistanceSmoothingElement
+    CreateAuxModelPart();
+
+    typename TLinearSolver::Pointer plinear_solver = LinearSolverFactory<TSparseSpace, TDenseSpace>().Create(
+        rParameters["linear_solver_settings"]);
+
+    auto p_builder_solver = Kratos::make_shared<ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver> >(plinear_solver);
+
+    CreateSolutionStrategy(plinear_solver, p_builder_solver);
+}
```

Comment:
these constructors contain a lot of copy paste
please forward the calls to the other constructors to avoid the copy-paste
check [how it is done in model_part.cpp](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/sources/model_part.cpp#L29-L56)

Reply:
Done.